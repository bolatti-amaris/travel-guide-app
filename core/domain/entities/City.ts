export interface City {
    id: number;
    key: string;
    name: string;
    nativeName: string;
    currency: string;
    language: string;
  }