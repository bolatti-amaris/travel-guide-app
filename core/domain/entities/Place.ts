export interface Place {
  type: 'monument' | 'restaurant';
  name: string;
  coordinates: [number, number];
}
