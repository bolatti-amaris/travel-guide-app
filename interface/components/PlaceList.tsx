import React from 'react';
import { View, Text } from 'react-native';
import { Place } from '@/core/domain/entities/Place';
import PlaceCard from './PlaceCard';

interface PlaceListProps {
  places: Place[];
}

const PlaceList = ({ places }: PlaceListProps) => {
  if (!places.length) return null;

  return (
    <View className="mt-4" testID="place-list">
      <Text
        testID="place-list-title"
        className="text-xl font-work-bold text-text-header dark:text-text-header-dark mb-2"
      >
        Places to visit:
      </Text>

      {places.map((place, idx) => (
        <PlaceCard key={idx} place={place} />
      ))}
    </View>
  );
};

export default PlaceList;
