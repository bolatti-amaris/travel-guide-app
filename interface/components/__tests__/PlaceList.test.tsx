import React from 'react';
import { render } from '@testing-library/react-native';
import { Place } from '@/core/domain/entities/Place';
import PlaceList from '../PlaceList';

jest.mock('../PlaceCard', () => {
  const { View } = require('react-native');
  return ({ testID }: { testID: string }) => <View testID={testID} />;
});

const mockPlaces: Place[] = [
  {
    name: 'Eiffel Tower',
    type: 'monument',
    coordinates: [48.8584, 2.2945],
  },
  {
    name: 'Louvre Museum',
    type: 'monument',
    coordinates: [48.8606, 2.3376],
  },
];

describe('PlaceList', () => {
  it('renders list wrapper and title', () => {
    const { getByTestId } = render(<PlaceList places={mockPlaces} />);
    expect(getByTestId('place-list')).toBeTruthy();
    expect(getByTestId('place-list-title')).toBeTruthy();
  });

  it('renders nothing if place list is empty', () => {
    const { queryByTestId } = render(<PlaceList places={[]} />);
    expect(queryByTestId('place-list')).toBeNull();
    expect(queryByTestId('place-list-title')).toBeNull();
  });

  it('matches snapshot', () => {
    const tree = render(<PlaceList places={mockPlaces} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
