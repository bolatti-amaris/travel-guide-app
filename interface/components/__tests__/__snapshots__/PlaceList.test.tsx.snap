// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceList matches snapshot 1`] = `
<View
  className="mt-4"
  testID="place-list"
>
  <Text
    className="text-xl font-work-bold text-text-header dark:text-text-header-dark mb-2"
    testID="place-list-title"
  >
    Places to visit:
  </Text>
  <View />
  <View />
</View>
`;
